### PR TITLE
haiku: add platform support.

### PR DESCRIPTION
**Pull request by jessicah**

### CODE REVIEW

File: src/errno.rs [hunk 1]
```diff
@@ -1910,6 +1925,172 @@ mod consts {
     }
 }
 
+#[cfg(target_os = "haiku")]
```
Comment (asomers) [inline src/errno.rs]: Nix does not define constants anymore.  Instead, all of these should be defined in the libc crate, and nix should reference libc.

Comment (jessicah) [reply src/errno.rs]: You mean how the [linux version](https://github.com/nix-rust/nix/blob/master/src/errno.rs#L529) is done? I had wondered why it was different to the other platforms.

File: src/errno.rs [hunk 2]
```diff
@@ -47,6 +47,12 @@ unsafe fn errno_location() -> *mut c_int {
     __errno()
 }
 
+#[cfg(target_os = "haiku")]
+unsafe fn errno_location() -> *mut c_int {
+    extern { fn _errnop() -> *mut c_int; }
```
Comment (Susurrus) [inline src/errno.rs]: This should also be declared in libc instead of here.

**Comment by jessicah** [comment]
Restricting to just the 'positive' would create unnecessarily complex `#[cfg(...)]` attributes, wouldn't it?
 Having to list every supported OS instead seems a bit cumbersome. If I go this route, I would prefer to list the targets one per line throughout. Would that be acceptable? Scrolling horizontally is a pain ;-)

File: src/errno.rs (hunk 1 above)
Comment (asomers) [reply src/errno.rs]: Yeah.  It looks like Somebody fixed the Linux definitions in change 3354ffe91dcc217c6ac62f88d5e54a1ea34e5f7a , but not other OS's.

File: src/fcntl.rs [hunk 3]
```diff
@@ -28,6 +28,7 @@ libc_bitflags!(
         O_ALT_IO;
         /// Open the file in append-only mode.
         O_APPEND;
+        #[cfg(not(target_os = "haiku"))]
```
Comment (Susurrus) [inline src/fcntl.rs]: These should actually go underneath the comment.

File: src/features.rs [hunk 4]
```diff
@@ -92,7 +92,7 @@ mod os {
     }
 }
 
-#[cfg(any(target_os = "macos", target_os = "freebsd", target_os = "dragonfly", target_os = "ios", target_os = "openbsd", target_os = "netbsd"))]
+#[cfg(any(target_os = "macos", target_os = "freebsd", target_os = "dragonfly", target_os = "ios", target_os = "openbsd", target_os = "netbsd", target_os = "haiku"))]
```
Comment (Susurrus) [inline src/features.rs]: Please line wrap these.

**Comment by Susurrus** [comment]
@jessicah You have any time to get around to resolving this? We'd love to have Haiku support!

**Comment by Susurrus** [comment]
@jessicah Wanted to ping you again on this.

**Comment by jessicah** [comment]
Sorry, yes, I will get back to it; laptop died, and haven't gotten around to setting up on my other machine.

**Comment by Susurrus** [comment]
We're about to cut a new release soon and I'd love to say that we fixed up Haiku support. So whenever you have time I'd love to merge this.

BTW what's your involvement with Haiku? You run it as a daily driver or just want to play with Rust on Haiku?

**Comment by jessicah** [comment]
Okay, I'll see if I can get to it this week.

I'm core committer/maintainer. Mainly working on rust so can run librespot on Haiku :p

**Comment by Susurrus** [comment]
@jessicah Any updates on this? We've had a bunch of syntax changes happen (like from #801), so this will likely need a good rebase. If you don't have time to do this in the near future, I'd rather close this and you can re-submit a PR whenever some time frees up for you.

**Comment by Susurrus** [comment]
I'm going to go ahead and close this, @jessicah, as it's been over 2 months since your last update. Please either reopen this PR or submit a new one that's been rebased on HEAD as a lot has changed since this was opened.

**Pull request closed without merging**